### PR TITLE
Doc for modules/optimizer.py

### PR DESCRIPTION
**Pull request by hieuchi911**
Add documentation for modules/optimizer.py

### CODE REVIEW

**Review by hjzccc — APPROVED** [review]
looks great

**Review by MrLYG — APPROVED** [review]
Good Example

File: ablator/modules/optimizer.py [hunk 1]
```diff
@@ -71,33 +143,163 @@ def get_optim_parameters(
...
+        Name of the optimizer.
+    arguments : OptimizerArgs
+        Arguments for the optimizer, specific to a certain type of optimizer.
+    """
     name: str
     arguments: OptimizerArgs
 
     def __init__(self, name, arguments: dict[str, ty.Any]):
+        """
+        Initializes the optimizer configuration. Add anny provided settings to the optimizer.
```
Comment (MrLYG) [inline ablator/modules/optimizer.py]:
```suggestion
        Initializes the optimizer configuration. Add any provided settings to the optimizer.
```

File: ablator/modules/optimizer.py [hunk 2]
```diff
@@ -12,6 +12,36 @@
 def get_parameter_names(model: torch.nn.Module, forbidden_layer_types: list[type]):
     """
     Returns the names of the model parameters that are not inside a forbidden layer.
```
Comment (fostiropoulos) [inline ablator/modules/optimizer.py]: this is incorrect

**Review by fostiropoulos — CHANGES_REQUESTED** [review]
can you please re-read the docs?

File: ablator/modules/optimizer.py (hunk 2 above)
Comment (hieuchi911) [reply ablator/modules/optimizer.py]: @fostiropoulos  How about this: "Recurse into the module and return parameter names of all submodules, excluding modules that are of any type defined in `forbidden_layer_types`."

For Return section: "The names of the parameters with the following format: `<submodule-name>.<parameter-name>`."

**Pull request merged**